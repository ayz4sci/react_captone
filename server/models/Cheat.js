import mongoose from 'mongoose';

const CheatSchema = {
    description: {type: String, required: true},
    command: {type: String, required: true},
    keywords: [{type: String}]
}

const Cheat = mongoose.model(
    'Cheat', CheatSchema, "cheats"
    );

module.exports = Cheat;