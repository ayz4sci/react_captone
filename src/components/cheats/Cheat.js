import React from 'react';
import PropTypes from 'prop-types';
import ClipboardEnabledText from '../utils/ClipboardEnabledText';

class Cheat extends React.Component{
    render(){
        return (
            <div className="cheat">
                <div>{this.props.cheat.description}</div>
                <ClipboardEnabledText content={this.props.cheat.command} />
            </div>
        )
    }
}

Cheat.propTypes = {
    cheat: PropTypes.shape({
        description: PropTypes.string.isRequired,
        command: PropTypes.string.isRequired,
        keywords: PropTypes.array
    }).isRequired
}

export default Cheat;