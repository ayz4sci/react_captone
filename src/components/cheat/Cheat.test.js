import React from 'react';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import Cheat from './Cheat';

it('renders correctly', () => {
    const props = { 
        cheat: {
            description: "Lol",
            command: "git command",
            keywords: []
        }
    };

    const wrapper = shallow(<Cheat {...props} />, {mode: 'shallow'})
    expect(toJson(wrapper)).toMatchSnapshot();
});