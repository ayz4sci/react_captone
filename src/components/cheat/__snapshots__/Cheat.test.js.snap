// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="cheat m-2"
>
  <div>
    Lol
  </div>
  <ClipboardEnabledText
    content="git command"
  />
</div>
`;
